src/main/java/nextstep/courses/domain/PaidRegistrationPolicy.java

```diff
@@ -0,0 +1,25 @@
+package nextstep.courses.domain;
+
+public class PaidRegistrationPolicy implements RegistrationPolicy {
+
+    private final Money sessionFee;
+    private final NaturalNumber maxStudentCount;
+
+    public PaidRegistrationPolicy(int sessionFee, int maxStudentCount) {
+        this.sessionFee = new Money(sessionFee);
+        this.maxStudentCount = new NaturalNumber(maxStudentCount);
+    }
+
+    @Override
+    public void validateRegistration(Session session, Money paymentAmount) {
+        if (maxStudentCount
+            .compareTo(session.getStudentCount()) <= 0) {
+            throw new IllegalArgumentException("강의 최대 수강 인원을 초과할 수 없습니다.");
+        }
```

Comment:
객체에 메시지를 보내는 방식으로 변경하면 더 객체지향적인 코드가 될 것 같아요!

```suggestion
        if (!session.canAcceptMoreStudents(maxStudentCount)) {
            throw new IllegalArgumentException("강의 최대 수강 인원을 초과할 수 없습니다.");
        }
```

이렇게 하면 `Session` 클래스에 수강 인원 체크 로직을 캡슐화할 수 있을것 같아요 🙂 
이는 객체의 상태를 직접 조회하는 대신 객체에게 책임을 위임하는 방식으로, 더 응집도 높은 설계가 될 것 같습니다! 👍

Reply:
@mskangg 
세션이 가져가는것 보단 RegistrationPolicy가 가져가는것이 좋아보여요.